### ISSUE TITLE
Kotlin 2.0

### ISSUE THREAD

**Issue by rcmkt** [yandex/yatagan#176]
Does **yatagan** support Kotlin 2.0 and k2 compiler? 
If not, will it planned?)

**Comment by Jeffset** [comment]
Depends on the backend one is using. KSP2 support is experimental, but it exists already. KAPT should just work.

**Comment by rcmkt** [comment]
Yeah, I'm interested in ksp2 support, any plans to make ksp stable? Just to understand the roadmap

**Comment by Jeffset** [comment]
The thing is, that KSP2 itself still has some issues for Java code generation. Yatagan may work well with KSP2 on the simple projects, on the more complex there may be some issues. So a lot of these issues will be out of my hands to fix. 

So, currently the best way to proceed would be you trying to adopt the framework and then report any found issues.

@byar @bacecek FYI

**Comment by Jeffset** [comment]
#180 #182

**Comment by Jeffset** [comment]
@bacecek I guess this can be closed as 1.6.0 is now released?

**Comment by bacecek** [comment]
> [@bacecek](https://github.com/bacecek) I guess this can be closed as 1.6.0 is now released?

Yeah, I guess so.

So, we have released 1.6.0 version, that includes several fixes for KSP2 mode. To ensure full compatibility, version 2.0.1 or higher of [google/ksp](https://github.com/google/ksp) is required.
